### PR TITLE
Pin icoscp and fix tests based on current retrieval

### PR DESCRIPTION
**Pull request by rt17603**
* **Summary of changes** (Bug fix, feature, docs update, ...)

The external `icoscp` package introduced new authentication as of 0.18.0 (see #612 ) which hasn't been integrated into openghg yet which means we need to pin to the previous version for now and ensure the current tests pass. This includes the following updates:
 - Pinned the icoscp package to the previous version in requirements.txt and environment.yml
 - Updated the tests/retrieve/icos tests to ensure they pass with the current configuration
 - ~Removed the icos mark for the tests/retrieve/icos tests which have mocked the retrieval so they are always run rather than just when the `--run-icos` flag is used with pytest. All retrieve icos tests were previously skipped due to this mark.~ *Would cause people's tests to fail if they don't have icoscp already pinned so will avoid for now.*

Note: that some of these tests have been built against external services (ICOS carbon portal accessed by icoscp) and so will need to be periodically reviewed / switch to using mockers exclusively for this.

This is needed to be able to continue with #1067 .

* **Please check if the PR fulfills these requirements**

- [x] Related to #612 but doesn't fully close this
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [x] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

Not needed
- ~Documentation and tutorials updated/added~

### CODE REVIEW

**Review by SutarPrasad — APPROVED** [review]
seems straightforward, good to go

**Comment by rt17603** [comment]
Will aim to merge once #1085 is resolved and tests are passing.

File: tests/retrieve/icos/test_retrieve_icos.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def test_icos_retrieve_skips_datalevel_1_csv_files():
     assert first_obs.metadata == expected_metadata
 
 
-@pytest.mark.icos
+# @pytest.mark.icos
```
Comment (rt17603) [inline tests/retrieve/icos/test_retrieve_icos.py:42]:
```suggestion
@pytest.mark.icos
```

**Review by brendan-m-murphy — APPROVED** [review]
Looks good!

**Pull request merged**